File: programmers/[고득점Kit]DFS,BFS/단어변환.py
# 단어변환

def solution(begin, target, words):
    new_begin = [begin]
    ans = 0

    if target not in words:
        return 0

    while (len(words) != 0):
        for b in new_begin:
            possible = []
            for word in words:
                cnt = 0
                for j in range(len(b)):
                    if b[j] != word[j]:
                        cnt += 1
                    if cnt == 2:
                        break
                if cnt == 1:
                    possible.append(word)
                    words.remove(word)
        ans += 1
        if target in possible:
            return ans
        else:
            new_begin = possible

    return 0

File: Algorithm-Swift/Algorithm-Swift/Programmers/[고득점Kit]BFS,DFS/단어변환.swift
//
//  단어변환.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/04/23.
//

import Foundation

func solution(_ begin:String, _ target:String, _ words:[String]) -> Int {
    var check = Array(repeating: false, count: words.count)
    var possible = [begin]
    var candidate: [String] = []
    var answer = 0
    
    if !words.contains(target){
        return 0
    }
    
    while check.filter({ !$0 }).count != 0 {
        for (idx,i) in words.enumerated(){
            if !check[idx]{
                let temp1 = Array(i)
                for j in possible{
                    var count = 0
                    let temp2 = Array(j)
                    for k in 0..<begin.count{
                        if temp1[k] == temp2[k]{
                            count += 1
                        }
                        if count == begin.count-1{
                            check[idx] = true
                            candidate.append(i)
                        }
                    }
                }
            }
        }
        answer += 1
        if candidate.contains(target){
            return answer
        }else{
            possible = candidate
        }
    }
    
    return 0
}